ReactCommon/cxxreact/Instance.h

```diff
@@ -62,9 +62,10 @@ class RN_EXPORT Instance {
 
   void setSourceURL(std::string sourceURL);
 
-  virtual void loadScriptFromString(std::unique_ptr<const JSBigString> bundleString,
-                            uint64_t bundleVersion, std::string bundleURL, bool loadSynchronously,
-                            std::string&& bytecodeFileName);
+  virtual void loadScriptFromString(
+      std::unique_ptr<const JSBigString> bundleString,
+      std::string bundleURL,
```

Comment:
>string bundleURL [](start = 11, length = 16)

why not pass it as a & or &&?

Reply:
I agree, but I'm not wanting to change existing interfaces for upstream RN.

There is a school of thought that you should accept move-only types by value instead of r-value, but I don't think that makes a lot of sense for something like string. They might just do a move internally, in which case that's no worse than taking in a const-ref and doing a copy.

---
In reply to: [342830990](https://github.com/microsoft/react-native/pull/187#discussion_r342830990) [](ancestors = 342830990)